packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -28,13 +25,13 @@ import 'icon_tree_shaker.dart';
 ///
 /// Returns a [Depfile] containing all assets used in the build.
 Future<Depfile> copyAssets(Environment environment, Directory outputDirectory, {
-  Map<String, DevFSContent> additionalContent,
-  @required TargetPlatform targetPlatform,
-  BuildMode buildMode,
+  Map<String, DevFSContent>? additionalContent,
+  required TargetPlatform targetPlatform,
+  BuildMode? buildMode,
```

Comment:
I went back and forth setting `buildMode` to `required` or nullable, it's nullable now since it isn't passed in a bunch of places.  The only place that seemed tricky was:
https://github.com/flutter/flutter/blob/783e1dd22ca357e1742b508516a5b56658e4aa46/packages/flutter_tools/lib/src/build_system/targets/assets.dart#L278

I erred on the side of keeping the existing behavior.

Reply:
makes sense